frontend/src/lib/Quiver.ts

```diff
@@ -599,30 +611,46 @@ but was expecting \`${JSON.stringify(expectedIndexTypes)}\`.
     return typeof indexName === "object" && indexName.kind === "range"
   }
 
+  /** Returns type for a single-index column or data column. */
+  public static getType(type: Type): IndexTypeName | string {
+    // For `PeriodIndex` and `IntervalIndex` types are kept in `numpy_type`,
+    // for the rest of the indexes in `pandas_type`.
+    const { pandas_type: pandasType, numpy_type: numpyType } = type
+
+    return pandasType === "object"
+      ? (numpyType as IndexTypeName)
+      : (pandasType as IndexTypeName | string)
+  }
+
```

Comment:
I think the assumption here is that if pandasType === "object", then numpyType should be non-null; otherwise pandasType should be non-null? If that's the case, I think we should explicitly check this, and throw an error immediately, rather than casting:

```ts
public static getTypeName(type: Type): IndexTypeName | string {
  // For `PeriodIndex` and `IntervalIndex` types are kept in `numpy_type`,
  // for the rest of the indexes in `pandas_type`.
  const { pandas_type: pandasType, numpy_type: numpyType } = type

  if (pandasType == null) {
    throw new Error(`pandas_type is unexpectedly null! (in ${type})`)
  }

  if (pandasType === "object") {
    if (numpyType == null) {
      throw new Error(`numpy_type is unexpectedly null! (in ${type})`)
    }

    return numpyType
  }

  return pandasType
}
```

(Alternately, this could return a `IndexTypeName | string | undefined`, if it's legal to have pandas_type or numpy_type be null)

Reply:
Refactored in a different way